### ISSUE TITLE
Bigram LM receives supervision only from numerator FSA?

### ISSUE THREAD

**Issue by vvwangvv** [jctian98/e2e_lfmmi#8]
Thanks for this great framework for unified e2e lfmmi training and inference!
I noticed that snowfall has updated the `MmiTrainingGraphCompiler` in [snowfall/training/mmi_graph.py:135](https://github.com/jctian98/e2e_lfmmi/blob/34b8056906631c6282bd2f0320764d645c33ca0d/snowfall/training/mmi_graph.py#L135) from
```python
ctc_topo_P_vec = k2.create_fsa_vec([ctc_topo_P.detach()])
```
to [a non-detach version](https://github.com/k2-fsa/snowfall/blob/911198817edc7b306265f32447ef8a7dc5cfa8f2/snowfall/training/mmi_graph.py#L136)
```python
ctc_topo_P_vec = k2.create_fsa_vec([self.ctc_topo_P])
```
Since in this repo the denominator FSA vec is detached, It seems that [bigram LM FSA parameters](https://github.com/jctian98/e2e_lfmmi/blob/34b8056906631c6282bd2f0320764d645c33ca0d/snowfall/warpper/warpper_mmi.py#L77) can only get supervision (gradient) from numerator FSA. 

I'm not sure if I missed something or the detach operation could be a problem.

Any help or explanation would be appreciated. Thanks!

**Comment by vvwangvv** [comment]
Found `detach` explained in [earlier commit of snowfall](https://github.com/k2-fsa/snowfall/pull/62#discussion_r550488505)
